### ISSUE TITLE
Build failure on macOS ARM64 with v22.0

### ISSUE THREAD

**Issue by carlocab** [protocolbuffers/protobuf#11996]
**What version of protobuf and what language are you using?**
Version: v22.0
Language: C++

**What operating system (Linux, Windows, ...) and version?**

macOS 13, 12, and 11 on ARM64. ~~My builds for the same OS versions on X64 are still running.~~ The same error does not appear on the same versions of macOS running on X64.

**What runtime / compiler are you using (e.g., python version or gcc version)**
- macOS 13: Apple Clang 14
- macOS 12: Apple Clang 14
- macOS 11: Apple Clang 13

**What did you do?**
Build Protobuf with CMake. Our CMake invocation is:
```
cmake -S . -B build -Dprotobuf_BUILD_LIBPROTOC=ON -Dprotobuf_BUILD_SHARED_LIBS=ON -Dprotobuf_INSTALL_EXAMPLES=ON -Dprotobuf_BUILD_TESTS=OFF -Dprotobuf_ABSL_PROVIDER=package -DCMAKE_CXX_STANDARD=17 -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/protobuf/22.0 -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DBUILD_TESTING=OFF -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk
```

**What did you expect to see**
A successful build.

**What did you see instead?**
We encountered the following linker error when linking `libprotoc`:
```
  Undefined symbols for architecture arm64:
    "google::protobuf::internal::VarintParseSlowArm32(char const*, unsigned int*, unsigned long long)", referenced from:
        char const* google::protobuf::internal::VarintParse<unsigned int>(char const*, unsigned int*) in plugin.pb.cc.o
    "google::protobuf::internal::VarintParseSlowArm64(char const*, unsigned long long*, unsigned long long)", referenced from:
        char const* google::protobuf::internal::VarintParse<unsigned long long>(char const*, unsigned long long*) in plugin.pb.cc.o
  ld: symbol(s) not found for architecture arm64
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  make[2]: *** [libprotoc.4.22.0.0.dylib] Error 1
  make[1]: *** [CMakeFiles/libprotoc.dir/all] Error 2
  make: *** [all] Error 2
```

**Anything else we should know about your project / environment**

Complete build logs available [here](https://github.com/Homebrew/homebrew-core/actions/runs/4211739550). On macOS 13-arm64, the error starts [here](https://github.com/Homebrew/homebrew-core/actions/runs/4211739550/jobs/7310336987#step:6:523).

We encountered this error while trying to package v22.0 for Homebrew at Homebrew/homebrew-core#123507.

**Comment by xtkoba** [comment]
This also happens to shared build for aarch64 Android.

A suggested patch: https://github.com/xtkoba/protobuf/commit/f91b227580326d7d67ad0cfa92a300b5ace55742

**Comment by ZhongRuoyu** [comment]
> A suggested patch: [xtkoba@f91b227](https://github.com/xtkoba/protobuf/commit/f91b227580326d7d67ad0cfa92a300b5ace55742)

If that's the case: [`PROTOBUF_EXPORT`](https://github.com/protocolbuffers/protobuf/blob/v3.22.0/src/google/protobuf/port_def.inc#L508) looks more appropriate. Controlled by [`cmake/libprotobuf.cmake`](https://github.com/protocolbuffers/protobuf/blob/main/cmake/libprotobuf.cmake#L30).

**Comment by carlocab** [comment]
This linker error seems specific to ARM64 then, and not only just an issue on macOS.

**Comment by acozzette** [comment]
I think @ZhongRuoyu is right that `PROTOBUF_EXPORT` should fix this. If anyone wants to send a pull request for this then we would be happy to review it.

**Comment by acozzette** [comment]
This should be fixed by 2ce56399e30db62e45869c6fd2d2bbacbb81a7ed and we will include that in the 22.1 release.
